src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD = 0.75f;
+    private static final int COEF = 2;
+
+    private Node<K, V>[] defaultTable;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        this.defaultTable = new Node[DEFAULT_INITIAL_CAPACITY];
+        this.threshold = (int)(DEFAULT_LOAD * DEFAULT_INITIAL_CAPACITY);
+    }
+
+    private static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = hashcode(key) % DEFAULT_INITIAL_CAPACITY;
+        if (defaultTable[index] == null) {
+            defaultTable[index] = new Node<>(key, value, null);
+            size++;
+            return;
+        }
+        Node<K, V> currentNode = defaultTable[index];
+        while ((currentNode.next != null) || (Objects.equals(currentNode.key, key))) {
+            if (Objects.equals(currentNode.key, key)) {
+                currentNode.value = value;
+                return;
+            }
+            currentNode = currentNode.next;
+        }
+        currentNode.next = new Node<>(key, value, null);
+        size++;
+        if (size == threshold) {
+            resize();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = Math.abs(hashcode(key)) % DEFAULT_INITIAL_CAPACITY;
```

Comment:
```suggestion
        int index = hashcode(key) % defaultTable.length;
```
Во первых, у тебя итак вернеться всегда положительное число.
Во вторых, вместо DEFAULT_INITIAL_CAPACITY используй длину массива.

Reply:
Мало вертатись, але в тест-кейсах були ключі з від'ємним хешкодом. Тому і використовував Math.abs, правда, тут це було зайвим, так як я його приписав у методі hashcode і не помітив, що тут не видалив.

Друге пофіксив.